src/Ordering.API/Application/DomainEventHandlers/OrderCancelledDomainEventHandler.cs

```diff
@@ -22,12 +22,12 @@ public partial class OrderCancelledDomainEventHandler
 
     public async Task Handle(OrderCancelledDomainEvent domainEvent, CancellationToken cancellationToken)
     {
-        OrderingApiTrace.LogOrderStatusUpdated(_logger, domainEvent.Order.Id, nameof(OrderStatus.Cancelled), OrderStatus.Cancelled.Id);
+        OrderingApiTrace.LogOrderStatusUpdated(_logger, domainEvent.Order.Id, OrderStatus.Cancelled);
 
         var order = await _orderRepository.GetAsync(domainEvent.Order.Id);
         var buyer = await _buyerRepository.FindByIdAsync(order.GetBuyerId.Value);
 
-        var integrationEvent = new OrderStatusChangedToCancelledIntegrationEvent(order.Id, order.OrderStatus.Name, buyer.Name, buyer.IdentityGuid);
+        var integrationEvent = new OrderStatusChangedToCancelledIntegrationEvent(order.Id, order.OrderStatus.ToString(), buyer.Name, buyer.IdentityGuid);
```

Comment:
Can we change the `OrderStatusChangedToCancelledIntegrationEvent` to take `OrderStatus` as an enum instead of a `string`?

Reply:
Hey Safia, thanks for your feedback on this! I'm looking into making this change, but I noticed that we serialize & persist the event data, so `OrderStatus` would end up being serialized as an `int` instead of a `string` by default. 

How do you want to handle that? I could throw a `[[JsonConverter(typeof(JsonStringEnumConverter))]` over the enum? Ex:

```csharp
public record OrderStatusChangedToAwaitingValidationIntegrationEvent : IntegrationEvent
{
    public int OrderId { get; }
    [JsonConverter(typeof(JsonStringEnumConverter))]
    public OrderStatus OrderStatus { get; }
    public string BuyerName { get; }
    public string BuyerIdentityGuid { get; }
    public IEnumerable<OrderStockItem> OrderStockItems { get; }

    public OrderStatusChangedToAwaitingValidationIntegrationEvent(
        int orderId, OrderStatus orderStatus, string buyerName, string buyerIdentityGuid,
        IEnumerable<OrderStockItem> orderStockItems)
    {
        OrderId = orderId;
        OrderStockItems = orderStockItems;
        OrderStatus = orderStatus;
        BuyerName = buyerName;
        BuyerIdentityGuid = buyerIdentityGuid;
    }
}
```

Reply:
@erinnmclaughlin I like that recommendation! I'm inclined to apply the attribute on the `public enum OrderStatus` definition itself so that it gets serialized consistently as a string throughout the codebase.

Reply:
@captainsafia didn't even realize that was an option 😅 that is much nicer. Done!